internal/proxy/proxy_test.go

```diff
@@ -306,6 +309,42 @@ func TestClientInitialization(t *testing.T) {
 				filepath.Join(testUnixSocketPathPg),
 			},
 		},
+		{
+			desc: "without TCP port or unix socket for non functional instance",
+			in: &proxy.Config{
+				Instances: []proxy.InstanceConnConfig{
+					{Name: "proj:region:fakeserver"},
+				},
+			},
+		},
+		{
+			desc: "with TCP port for non functional instance",
```

Comment:
We probably also want:

1. a test case with a working TCP socket and a failing Unix socket
2. a test case with a working TCP socket and a failing TCP socket (think two TCP sockets both configured to listen on the same port).

Reply:
Added these two cases. Also let proxy exit if the tcp socket spec is wrong.